compiler/ras/DebugCounter.md

```diff
@@ -0,0 +1,126 @@
+# Debug Counter
+
+A debug counter is a facility for adding instrumentation to jitted code to count arbitrary things.
```

Comment:
There is also an incStaticDebugCounter which can be used to count things at compile-time. Perhaps a better introduction would be to say debug counters are a general infrastructure for counting things. There are two flavours - static and dynamic. Static counters do not add anything to the code generated by the compiler, but server to count events at compile-time while dynamic counters are incremented at runtime and are used to count events in code generated by the compiler.